common/src/main/java/com/google/auto/common/MoreTypes.java

```diff
@@ -314,7 +336,7 @@ private static boolean equal(
     // ExecutableType.
     @SuppressWarnings("TypesEquals")
     boolean equal = a.equals(b);
-    if (equal && !(a instanceof ExecutableType)) {
+    if (equal && (a.getKind() != TypeKind.EXECUTABLE)) {
```

Comment:
I don't think the extra parentheses are necessary and I'd suggest removing them.

Reply:
No problem. Done.